roles/mongodb_linux/tasks/main.yml

```diff
@@ -65,7 +59,7 @@
     value: 0
     state: present
     reload: yes
-  when: d.stat.exists == False
+  when: ansible_facts.virtualization_type == 'docker'
```

Comment:
Logic is backtofront here.

Reply:
Isn't it desired? What would be the alternative I wonder :thinking:

Reply:
So, is_docker file will be created when you are using docker virtualization . Is my understanding correct?

Reply:
Yes

Reply:
So probably something should happen in `prepare.yml`? :thinking:

Reply:
No, we're getting rid of that. That's the point in this PR. Jacob's idea to use the fact is much cleaner than creating the is_docker file in prepare.yml. You were right to delete this file it's just you're testing for the opposite (wrong) thing here.

Reply:
One thing confuses me a bit - so if we completely remove the prepare.yml then when the control goes to ./tests/ansible-operator/molecule/kind/molecule.yml line no. 16 , then it will search for existence of prepare.yml and will throw an error. There are other prepare.yml in other roles which doesn't just have is_docker check. So, I think the code would break, that's why I had only commented out the tasks in concerned prepare.yml in my PR

Reply:
The file ./tests/ansible-operator/molecule/kind/molecule.yml  is irrelevant here as you're not working on it. prepare.yml (and other playbook names) are default playbook names, with the default setup molecule will run them in the correct order if they exist but just carry on happily if they don't exist. If memory serves correctly none of the role molecule tests reference prepare.yml so no problem. If these are explicitly mentioned in molecule.yml it might throw an error, if they don't exist, but then we can just remove that reference. 

If a prepare.yml only has a single task that we want to remove then we probably want to delete it. I don't think molecule would be happy with a playbook containing no tasks.

Instead of just guessing what you think will happen run the molecule tests. This is one of the reasons why they are there. They will tell you if you've broken anything and this is much faster than waiting for feedback from us.

Cheers,

Rhys